glean_parser/schemas/metrics.2-0-0.schema.yaml

```diff
@@ -685,3 +685,20 @@ additionalProperties:
                         Built-in pings are not allowed."
                       pattern:
                         "^(metrics|baseline|events|deletion-request|default|glean_.*)$"
+
+      -
+        if:
+          properties:
+            type:
+              enum:
+                - datetime
+                - timing_distribution
+                - timespan
+          required:
+            - unit
```

Comment:
Oh wow, I had to read the note under [this part of the reference](https://json-schema.org/understanding-json-schema/reference/conditionals.html#if-then-else) to understand what's going on here.

So, if the type is one of the three _and_ there is a `unit`, _then_ we require a `time_unit` with a helpful message.

Which suggests to me that this will fail to error if someone includes both a `unit` _and_ a `time_unit` field? We may need a test for that.

Reply:
Yeah, it definitely took my brain a little while to understand what this is doing.  Maybe a comment?